File: polyglot/windows/dotnet/SeisWindowsPlatform.fs
namespace Seis.Windows

type Capability =
    { Id: string
      Languages: string list
      QualityGates: string list
      OfflineHelper: bool
      RemoteBridge: bool }

module SeisWindowsPlatform =
    let capability =
        { Id = "windows-dotnet-fsharp"
          Languages = [ "F#"; "C#"; "Visual Basic"; "PowerShell" ]
          QualityGates =
            [ "dotnet_readiness"
              "windows_path_safety"
              "permission_scope"
              "offline_fallback"
              "event_log_awareness" ]
          OfflineHelper = true
          RemoteBridge = true }

    let isReady capability =
        capability.Languages.Length >= 4
        && capability.QualityGates.Length >= 5
        && capability.OfflineHelper
        && capability.RemoteBridge

File: polyglot/windows/dotnet/SeisWindowsPlatform.vb
Namespace Seis.Windows
    Public Module SeisWindowsPlatform
        Public Function Languages() As String()
            Return New String() {"Visual Basic", "C#", "F#", "PowerShell"}
        End Function

        Public Function IsReadyForSeisAgent(ByVal qualityGateCount As Integer,
                                           ByVal offlineHelper As Boolean,
                                           ByVal remoteBridge As Boolean) As Boolean
            Return qualityGateCount >= 5 AndAlso offlineHelper AndAlso remoteBridge
        End Function
    End Module
End Namespace
